// react-stormpath/lib/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.UserComponent = exports.UserField = exports.VerifyEmailView = exports.ResetPasswordForm = exports.RegistrationForm = exports.LoginForm = exports.LogoutLink = exports.LoginLink = exports.NotAuthenticated = exports.Authenticated = exports.AuthenticatedRoute = exports.LogoutRoute = exports.LoginRoute = exports.HomeRoute = exports.constants = exports.actions = exports.Router = exports.Context = undefined;
exports.init = init;

var _app = require('./app');

var _app2 = _interopRequireDefault(_app);

var _Context2 = require('./Context');

var _Context3 = _interopRequireDefault(_Context2);

var _Router2 = require('./components/Router');

var _Router3 = _interopRequireDefault(_Router2);

var _UserActions = require('./actions/UserActions');

var _UserActions2 = _interopRequireDefault(_UserActions);

var _UserConstants = require('./constants/UserConstants');

var _UserConstants2 = _interopRequireDefault(_UserConstants);

var _HomeRoute2 = require('./components/HomeRoute');

var _HomeRoute3 = _interopRequireDefault(_HomeRoute2);

var _LoginRoute2 = require('./components/LoginRoute');

var _LoginRoute3 = _interopRequireDefault(_LoginRoute2);

var _LogoutRoute2 = require('./components/LogoutRoute');

var _LogoutRoute3 = _interopRequireDefault(_LogoutRoute2);

var _AuthenticatedRoute2 = require('./components/AuthenticatedRoute');

var _AuthenticatedRoute3 = _interopRequireDefault(_AuthenticatedRoute2);

var _Authenticated2 = require('./components/Authenticated');

var _Authenticated3 = _interopRequireDefault(_Authenticated2);

var _NotAuthenticated2 = require('./components/NotAuthenticated');

var _NotAuthenticated3 = _interopRequireDefault(_NotAuthenticated2);

var _LoginLink2 = require('./components/LoginLink');

var _LoginLink3 = _interopRequireDefault(_LoginLink2);

var _LogoutLink2 = require('./components/LogoutLink');

var _LogoutLink3 = _interopRequireDefault(_LogoutLink2);

var _LoginForm2 = require('./components/LoginForm');

var _LoginForm3 = _interopRequireDefault(_LoginForm2);

var _RegistrationForm2 = require('./components/RegistrationForm');

var _RegistrationForm3 = _interopRequireDefault(_RegistrationForm2);

var _ResetPasswordForm2 = require('./components/ResetPasswordForm');

var _ResetPasswordForm3 = _interopRequireDefault(_ResetPasswordForm2);

var _VerifyEmailView2 = require('./components/VerifyEmailView');

var _VerifyEmailView3 = _interopRequireDefault(_VerifyEmailView2);

var _UserField2 = require('./components/UserField');

var _UserField3 = _interopRequireDefault(_UserField2);

var _UserComponent2 = require('./components/UserComponent');

var _UserComponent3 = _interopRequireDefault(_UserComponent2);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.Context = _Context3.default;
exports.Router = _Router3.default;
exports.actions = _UserActions2.default;
exports.constants = _UserConstants2.default;
exports.HomeRoute = _HomeRoute3.default;
exports.LoginRoute = _LoginRoute3.default;
exports.LogoutRoute = _LogoutRoute3.default;
exports.AuthenticatedRoute = _AuthenticatedRoute3.default;
exports.Authenticated = _Authenticated3.default;
exports.NotAuthenticated = _NotAuthenticated3.default;
exports.LoginLink = _LoginLink3.default;
exports.LogoutLink = _LogoutLink3.default;
exports.LoginForm = _LoginForm3.default;
exports.RegistrationForm = _RegistrationForm3.default;
exports.ResetPasswordForm = _ResetPasswordForm3.default;
exports.VerifyEmailView = _VerifyEmailView3.default;
exports.UserField = _UserField3.default;
exports.UserComponent = _UserComponent3.default;

// When not using ES6, enable people to use ReactStormpath.init()
// instead of ReactStormpath.default.init().

function init() {
  _app2.default.init.apply(_app2.default, arguments);
};

exports.default = _app2.default;
